daemon/algod/api/server/v2/handlers.go

```diff
@@ -588,6 +589,11 @@ func (v2 *Handlers) AbortCatchup(ctx echo.Context, catchpoint string) error {
 // TealCompile compiles TEAL code to binary, return both binary and hash
 // (POST /v2/teal/compile)
 func (v2 *Handlers) TealCompile(ctx echo.Context) error {
+	// return early if teal compile is not allowed in node config
+	if (! v2.Node.Config().EnableTealService) {
+		return ctx.String(http.StatusLocked, "/teal/compile disabled by node config.")
```

Comment:
I'm not sure the implications of "StatusLocked", a 404 is probably fine. It looks like there is a shortcut for this, so you can return `echo.ErrNotFound`